script/modelling/n-queens.jl

```diff
@@ -0,0 +1,66 @@
+#' ---
+#' title: N-Queens
+#' ---
+
+#' **Originally Contributed by**: Matthew Helm
+
+#' The N-Queens problem involves placing N queens on an N x N chessboard such that none of the queens attacks another. In chess, a 
+#' queen can move vertically, horizontally, and diagonally so there cannot be more than one queen on any given row, column, or 
+#' diagonal.
+
+#' <img src="img/n_queens4.png" style="width: auto; height: auto" alt="4 Queens">
+
+#' *Note that none of the queens above are able to attack any other as a result of their careful placement.*
+
+using GLPK
+using JuMP
+using LinearAlgebra
+
+# N-Queens
+N = 8
+
+model = Model(GLPK.Optimizer);
+
+#' Next, let's create an N x N chessboard of binary values. 0 will represent an empty space on the board and 1 will represent a 
+#' space occupied by one of our queens:
+
+@variable(model, x[i=1:N, j=1:N], Bin)
+
+#' Now we can add our constraints:
+
+# There must be exactly one queen in a given row/column
+for i=1:N
+    @constraint(model, sum(x[i, :]) == 1)
+    @constraint(model, sum(x[:, i]) == 1)
+end
+
+#' That's it! We are ready to put our model to work and see if it is able to find a feasible solution:
+
+optimize!(model)
+
+#' We can now review the solution that our model found:
+
+solution = value.(x)
+
+#' <img src="img/n_queens.png" style="width: auto; height: auto" alt="4 Queens">
+
+#' ** Bonus content:**
+
+#' Here is the code to plot your solutions with Makie.jl:
+
+using Base.Iterators: repeated
+using Makie
```

Comment:
I see you are using Makie there. I think we should try to homogeneize the plot backend we are using. 
E.g. it looks like `script/modelling/rocket_control.jl` is using Gadfly. 
As plotting libraries could be pretty heavy dependencies, I would be in favor of using [RecipesBase.jl](https://github.com/JuliaPlots/RecipesBase.jl). But that would deserve more careful thinking

Reply:
I wasn't sure if it was even appropriate to include this as it is unrelated to the JuMP tutorial. I decided though that since I wrote the code to generate the visuals, I might as well make it available to anyone following along with the tutorial. What do you think about either removing it or including it as a comment or markdown?

Reply:
Another solution would be to print the solution directly in the REPL, with an appropriate `@printf` formatting. I think displaying the solution would help a lot the understanding.

Reply:
Right now the solution will print as a matrix and the chessboards are static .png images - they aren't actually generated with the code. I'll remove the Makie code and re-submit.

Reply:
I converted the values in the solution matrix to ```Int``` from ```Float``` so that the solutions matrix is easier to view in the REPL/Jupyter output.